src/main/java/io/supertokens/multitenancy/MultitenancyHelper.java

```diff
@@ -84,80 +85,81 @@ private TenantConfig[] getAllTenantsFromDb() throws StorageQueryException {
         return StorageLayer.getMultitenancyStorage(main).getAllTenants();
     }
 
-    public void refreshTenantsInCoreIfRequired(boolean reloadAllResources) {
+    public List<TenantIdentifier> refreshTenantsInCoreIfRequired(boolean reloadAllResources) {
         try {
-            main.getResourceDistributor().withResourceDistributorLock(() -> {
+            return main.getResourceDistributor().withResourceDistributorLock(() -> {
                 try {
                     TenantConfig[] tenantsFromDb = getAllTenantsFromDb();
 
-                    boolean hasChanged = false;
-                    if (tenantsFromDb.length != tenantConfigs.length) {
-                        hasChanged = true;
-                    } else {
-                        Map<TenantIdentifier, TenantConfig> fromDb = new HashMap<>();
-                        for (TenantConfig t : tenantsFromDb) {
-                            fromDb.put(t.tenantIdentifier, t);
-                        }
-                        for (TenantConfig t : this.tenantConfigs) {
-                            TenantConfig fromDbConfig = fromDb.get(t.tenantIdentifier);
-                            if (!t.deepEquals(fromDbConfig)) {
-                                hasChanged = true;
-                                break;
-                            }
```

Comment:
need to propagate changes to base app tenant to all tenants...

Reply:
Done